src/test/java/de/siegmar/logbackgelf/pool/SimpleObjectPoolTest.java

```diff
@@ -93,6 +97,79 @@ public void maxLastBorrowed() throws InterruptedException {
         pool.returnObject(o3);
     }
 
+    @Test
+    void availableObjectsExhausted() throws InterruptedException {
+        final SimpleObjectPool<MyPooledObject> pool =
+                new SimpleObjectPool<>(factory, 1, 100, 100, 0);
+
+        final Object o1 = pool.borrowObject();
+        assertNotNull(o1);
+
+        assertThrows(IllegalStateException.class, pool::borrowObject,
+                "Couldn't acquire connection from pool");
+    }
+
+    @Test
+    void validatePoolSize() {
+        assertThrows(IllegalArgumentException.class, () ->
+                new SimpleObjectPool<>(factory, 0, 100, 100, 0),
+                "poolSize must be > 0");
+    }
+
+    @Test
+    void infinitePollWaitTime() throws InterruptedException {
+        final SimpleObjectPool<MyPooledObject> pool =
+                new SimpleObjectPool<>(factory, 1, -1, 100, 0);
+
+        final MyPooledObject o1 = pool.borrowObject();
+        final long before = System.currentTimeMillis();
+
+        new Thread(() -> {
+            try {
+                Thread.sleep(200);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            pool.returnObject(o1);
+        }).start();
+
+        final MyPooledObject o2 = pool.borrowObject();
+        final long after = System.currentTimeMillis();
+
+        assertNotNull(o2);
+        assertTrue(after - before > 190);
+    }
+
+    @Test
+    void infiniteLifeTime() throws InterruptedException {
+        final SimpleObjectPool<MyPooledObject> pool =
+                new SimpleObjectPool<>(factory, 1, 100, -1, -1);
+
+        final MyPooledObject o1 = pool.borrowObject();
+        pool.returnObject(o1);
+
+        Thread.sleep(200);
```

Comment:
I don't think is really required. I added it as a precaution, to avoid a false positive in case of the implementation changes. I can remove the sleep, or just reduce it to 20 milliseconds. What do you think?

Reply:
Let's remove it and only add it with a comment on why it is needed (which I doubt).

Reply:
I just removed it